File: test/analytics.test.js
import test from 'node:test';import assert from 'node:assert/strict';import {periodBounds,buildReport,toIso} from '../analytics.js';
test('calcola i confini mensili',()=>{const [a,b]=periodBounds('month',new Date(2026,1,12,12));assert.equal(toIso(a),'2026-02-01');assert.equal(toIso(b),'2026-02-28');});
test('conta più attività nella stessa giornata',()=>{const data={operators:[{id:'o1',name:'Mario'}],activities:[{id:'a',name:'A'},{id:'b',name:'B'}],assignments:{'o1|2026-02-03':['a','b'],'o1|2026-02-04':['a']}};const r=buildReport(data,'month',new Date(2026,1,12,12));assert.equal(r.totalAssignments,3);assert.equal(r.byOperator[0].activeDays,2);assert.equal(r.byOperator[0].percentages.a,2/3);});

File: test/analytics_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:squadra_tracker/analytics.dart';
import 'package:squadra_tracker/models.dart';

void main() {
  test('conta più attività nello stesso giorno', () {
    final data = AppData(
      operators: [const OperatorModel(id: 'o1', name: 'Mario')],
      activities: const [
        ActivityModel(id: 'a', name: 'A', color: 0),
        ActivityModel(id: 'b', name: 'B', color: 0),
      ],
      assignments: {
        'o1|2026-02-03': ['a', 'b'],
        'o1|2026-02-04': ['a'],
      },
    );
    final row = buildReport(
      data,
      PeriodType.month,
      DateTime(2026, 2, 12),
    ).single;
    expect(row.total, 3);
    expect(row.activeDays, 2);
    expect(row.percent('a'), closeTo(2 / 3, .0001));
  });
  test('calcola il mese intero', () {
    final (a, b) = periodBounds(PeriodType.month, DateTime(2026, 2, 12));
    expect(isoDate(a), '2026-02-01');
    expect(isoDate(b), '2026-02-28');
  });
}

